Source/WebKit/UIProcess/API/wpe/WPEView.cpp

```diff
@@ -363,6 +363,18 @@ WebKitWebResourceLoadManager* View::webResourceLoadManager()
     return m_client->webResourceLoadManager();
 }
 
+#if ENABLE(FULLSCREEN_API)
+void View::enterFullScreen()
+{
+    m_client->enterFullScreen(*this);
+}
+
+void View::exitFullScreen()
+{
+    m_client->exitFullScreen(*this);
+}
+#endif
```

Comment:
I don't think these are really needed. Wouldn't it be better to call `m_client->enter/exitFullScreen` from `View::setFullScreen`? Otherwise, it's really unclear why you have to separately call `setFullScreen` and `enterFullScreen` or `exitFullScreen` on the same WPEView.

Reply:
In my original implementation, I used `View::setFullScreen()` only.  But then I noticed that for every `API::ViewClient` method there is a corresponding method in `WKWPE::View`:
https://github.com/WebKit/WebKit/blob/13fa2acdd4842002cebe0ba83ca272cb4213f391/Source/WebKit/UIProcess/API/wpe/WPEView.h#L76-L82
I thought that it might be a good idea to keep this pattern. But I'm not sure. If you still think we don't need these separate API calls I can get rid of them. So, what would you say?

Reply:
I decided not to change the existing API and do what you originally suggested.